src/rest_vol.c

```diff
@@ -1918,35 +1931,41 @@ RV_copy_object_URI_callback(char *HTTP_response, void *callback_data_in, void *c
  *
  *              While the type of the target object is unknown, this
  *              function will keep locating the group that the link to the
- *              target object is contained within, resolving soft links as
- *              it is processing, until it locates the final hard link to
+ *              target object is contained within, resolving soft links
+ *              by repeated requests to the server,
+ *              until it locates the final hard link to
  *              the target object, at which point it will set the resulting
- *              type. Once the type of the target object is known, we can
- *              directly make the HTTP GET request to retrieve the object's
- *              information.
+ *              type. Once the type of the target object is known, the
+ *              HTTP GET request to directly retrieve the object's
+ *              information is made.
  *
  * Return:      Non-negative on success, negative on failure
  *
  * Programmer:  Jordan Henderson
  *              November, 2017
  */
 htri_t
-RV_find_object_by_path1(RV_object_t *parent_obj, const char *obj_path, H5I_type_t *target_object_type,
-                        herr_t (*obj_found_callback)(char *, void *, void *), void *callback_data_in,
-                        void *callback_data_out)
+RV_find_object_by_path(RV_object_t *parent_obj, const char *obj_path, H5I_type_t *target_object_type,
+                       herr_t (*obj_found_callback)(char *, void *, void *), void *callback_data_in,
+                       void *callback_data_out)
 {
-    RV_object_t *external_file         = NULL;
-    hbool_t      is_relative_path      = FALSE;
-    size_t       host_header_len       = 0;
-    char        *host_header           = NULL;
-    char        *path_dirname          = NULL;
-    char        *tmp_link_val          = NULL;
-    char        *url_encoded_link_name = NULL;
-    char        *url_encoded_path_name = NULL;
-    char         request_url[URL_MAX_LENGTH];
-    long         http_response;
-    int          url_len   = 0;
-    htri_t       ret_value = FAIL;
+    RV_object_t       *external_file    = NULL;
+    hbool_t            is_relative_path = FALSE;
+    size_t             host_header_len  = 0;
+    H5L_info2_t        link_info;
+    char              *url_encoded_link_name = NULL;
+    char              *host_header           = NULL;
+    char              *path_dirname          = NULL;
+    char              *tmp_link_val          = NULL;
+    char              *url_encoded_path_name = NULL;
+    char              *ext_filename          = NULL;
+    char              *ext_obj_path          = NULL;
+    char               request_url[URL_MAX_LENGTH];
+    long               http_response;
+    int                url_len = 0;
+    server_api_version version = parent_obj->domain->u.file.server_version;
```

Comment:
This should be moved down so that we don't try to de-reference `parent_obj` before we've checked that it's non-NULL.